### DISCUSSION TITLE
Output of the auth code

### DISCUSSION THREAD

**Discussion by jcs090218** [znck/grammarly#281] [General]
Original comment in https://github.com/grammarly/grammarly-for-developers/discussions/18#discussioncomment-2846782

---

Hi, I have implemented the [lsp-grammarly](https://github.com/emacs-grammarly/lsp-grammarly) (Emacs client), but I am have issue to login with just Emacs (without vscode installed).

The old [method 2](https://github.com/emacs-grammarly/lsp-grammarly#-method-2-login-with-emacs-a-bit-complicated) will print the auth code in the browser console, but now it won't:

![](https://raw.githubusercontent.com/emacs-grammarly/lsp-grammarly/master/etc/login/external-handler.png)

Is there a way to login (connect) to the account without using VSCode? 😕

**Comment by chapeiro** [comment]
Hey @jcs090218 , I think you should push the full url to the language server instead of only the code (if I am reading the emac plugin code correctly). I followed a similar approach to yours for the sublime plugin: https://github.com/sublimelsp/LSP-Grammarly/blob/caa3396ab885ece3c0a12a9ff48bab8ff86862b0/plugin.py#L54  (although I am still waiting for a separate client ID so that we do not piggyback on vscode's id).

**Reply by jcs090218** [reply]
Thanks for the suggestion!

[lsp-grammarly]() is the Emacs client, and uses [znck/grammarly](https://github.com/znck/grammarly)'s (this repo) language server.

**Reply by znck** [reply]
`@grammarly/sdk` package is in very early stage of development (it's experimental and may change in future). I am putting a case to invite developers building language server clients for early access.
